scripts/nix-profile.sh.in

```diff
@@ -1,4 +1,4 @@
-if [ -n "$HOME" ] && [ -n "$USER" ]; then
+if [ -n "$HOME" ] && [ -n "$USER" ] && ! USER=$(id -u -n); then
```

Comment:
I might be wrong (conditional expressions are always wonderfully tricky to the human mind), but it doesn't seem to do what you intent. The third condition will only be reached if `$USER` is already set and non empty.

That being said, `$USER` isn't used in this script (since c9159f86cc9a2fc07e2ab1217c2d8a8824123df4, so it's quite old) so maybe we could drop the condition altogether